### DISCUSSION TITLE
Template Repositories - Additional configuration options

### DISCUSSION THREAD

**Discussion by chrisreddington** [community/community#10768] [Other Feature Feedback, Questions, & Ideas]
I've created a [Template Repository](https://github.com/CloudWithChris/base-template) in my organisation, making customisations beyond the code in the repo such as - 

- Enabling GitHub Discussions
- Creating my own types of 'default' labels for GitHub issues
- Enabling a branch policy on main

When using a repository as a template, it would be brilliant if there was an additional set of steps in place to determine what you want to 'cookie cut'. For example, is it just the repo code? Is it the branch policies too? Is it the custom labels that you have configured? Are there bots/apps assigned that you want to carry forward?

**Comment by dgteixeira** [comment]
This is something that would be very helpfull!

**Comment by JamesAskew** [comment]
+1

I would like to see the option to use a specific branch of the template.

For my use case, I want to test changes before merging them in to the main branch of the template repo. At the moment the only option is to fork the template, test the changes on the forked repo then merge.

I specifically want to do it through the cli, but my feature request there got closed: https://github.com/cli/cli/issues/5868
